code/modules/mob/living/simple_animal/bot/SuperBeepsky.dm

```diff
@@ -0,0 +1,152 @@
+	weapon.attack(C, src)
+	playsound(src, 'sound/weapons/blade1.ogg', 50, TRUE, -1)
+	if(C.stat == DEAD)
+		addtimer(CALLBACK(src, .proc/update_icon), 2)
+		back_to_idle()
+
+
+/mob/living/simple_animal/bot/secbot/grievous/handle_automated_action()
+	if(!on)
+		return
+	switch(mode)
+		if(BOT_IDLE)		// idle
+			update_icon()
+			walk_to(src,0)
+			look_for_perp()	// see if any criminals are in range
+			if(!mode && auto_patrol)	// still idle, and set to patrol
+				mode = BOT_START_PATROL	// switch to patrol mode
+		if(BOT_HUNT)		// hunting for perp
+			update_icon()
+			playsound(src,'sound/effects/beepskyspinsabre.ogg',100,TRUE,-1)
+			// general beepsky doesn't give up so easily, jedi scum
+			if(frustration >= 20)
+				walk_to(src,0)
+				back_to_idle()
+				return
+			if(target)		// make sure target exists
+				if(Adjacent(target) && isturf(target.loc))	// if right next to perp
+					target_lastloc = target.loc //stun_attack() can clear the target if they're dead, so this needs to be set first
+					stun_attack(target)
+					anchored = TRUE
+					return
+				else								// not next to perp
+					var/turf/olddist = get_dist(src, target)
+					walk_to(src, target,1,4)
+					if((get_dist(src, target)) >= (olddist))
+						frustration++
+					else
+						frustration = 0
+			else
+				back_to_idle()
+
+		if(BOT_START_PATROL)
+			look_for_perp()
+			start_patrol()
+
+		if(BOT_PATROL)
+			look_for_perp()
+			bot_patrol()
+
+/mob/living/simple_animal/bot/secbot/grievous/look_for_perp()
+	anchored = FALSE
+	var/judgement_criteria = judgement_criteria()
+	for (var/mob/living/carbon/C in view(7,src)) //Let's find us a criminal
+		if((C.stat) || (C.handcuffed))
+			continue
+
+		if((C.name == oldtarget_name) && (world.time < last_found + 100))
+			continue
+
+		threatlevel = C.assess_threat(judgement_criteria, weaponcheck=CALLBACK(src, .proc/check_for_weapons))
+
+		if(!threatlevel)
+			continue
+
+		else if(threatlevel >= 4)
+			target = C
+			oldtarget_name = C.name
+			speak("Level [threatlevel] infraction alert!")
+			playsound(src, pick('sound/voice/bcriminal.ogg', 'sound/voice/bjustice.ogg', 'sound/voice/bfreeze.ogg'), 50, FALSE)
+			playsound(src,'sound/weapons/saberon.ogg',50,TRUE,-1)
+			visible_message("[src] ignites his energy swords!")
+			icon_state = "grievous-c"
+			visible_message("<b>[src]</b> points at [C.name]!")
+			mode = BOT_HUNT
+			INVOKE_ASYNC(src, .proc/handle_automated_action)
+			break
+		else
+			continue
+
+
+/mob/living/simple_animal/bot/secbot/grievous/explode()
+
+	walk_to(src,0)
+	visible_message("<span class='boldannounce'>[src] lets out a huge cough as it blows apart!</span>")
+	var/atom/Tsec = drop_location()
+
+	var/obj/item/bot_assembly/secbot/Sa = new (Tsec)
+	Sa.build_step = 1
+	Sa.add_overlay("hs_hole")
+	Sa.created_name = name
+	new /obj/item/assembly/prox_sensor(Tsec)
+
+	if(prob(50))
+		drop_part(robot_arm, Tsec)
+
+	do_sparks(3, TRUE, src)
+	qdel(weapon)
```

Comment:
QDEL_NULL

Reply:
Or just remove the line entirely since you qdel(src) below and /Destroy() already does it